sale_timesheet_line_exclude/models/account_analytic_line.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2018-2019 Brainbean Apps
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    exclude_from_sale_order = fields.Boolean(
+        string='Exclude from Sale Order',
+        help=(
+            'Checking this would exclude this timesheet entry from Sale Order'
+        ),
+    )
+
+    @api.onchange('task_id', 'employee_id')
+    def _onchange_task_id_employee_id(self):
+        if self.project_id:  # timesheet only
+            self.so_line = self._timesheet_get_sale_line()
+
+    @api.onchange('exclude_from_sale_order')
+    def _onchange_exclude_from_sale_order(self):
+        if self.project_id:  # timesheet only
```

Comment:
If you change `project_id` to `False`, standard code remove the sales order line?

Reply:
Nope, see https://github.com/odoo/odoo/blob/2efcefb50410e234faf777bdf88d879368907915/addons/sale_timesheet/models/account.py#L41

Reply:
I think that should be fixed upstream then

Reply:
Yet it won't be accepted in v12 anyhow, so this code should be consistent with base?

Reply:
Not sure... manual analytic account lines without project_id are filled with sales order line?

Reply:
It will be erased on `write()` anyhow, since there can not be task_id without project. I can rewrite it to explicitly set so_line to False if there's no project while in UI (in onchange)

Reply:
OK, understood. I think there's no need for that and it's not related to this module.

Reply:
Agreed